src/main/java/chess/domain/Board.java

```diff
@@ -0,0 +1,89 @@
+package chess.domain;
+
+import chess.domain.piece.Direction;
+import chess.domain.piece.Empty;
+import chess.domain.piece.Pawn;
+import chess.domain.piece.Piece;
+import chess.domain.piece.Team;
+import chess.domain.player.Player;
+import java.util.Map;
+
+public class Board {
+
+    private final Map<Point, Piece> board;
+
+    public Board(Map<Point, Piece> board) {
+        this.board = board;
+    }
+
+    public void move(Player player, Point currentPoint, Point destination) {
+        Piece currentPiece = board.get(currentPoint);
+
+        validateMove(player, currentPoint, destination, currentPiece);
+
+        movePiece(currentPoint, destination, currentPiece);
+    }
+
+    private void movePiece(Point currentPoint, Point destination, Piece currentPiece) {
+        board.put(currentPoint, new Empty(Team.EMPTY));
+        board.put(destination, currentPiece);
+    }
+
+    private void validateMove(Player player, Point currentPoint, Point destination, Piece currentPiece) {
+        validateMyPiece(player, currentPiece);
+        validateMovablePoint(currentPoint, destination, currentPiece);
+        validateDestination(player, destination);
+        validateMovableRoute(currentPoint, destination);
+        validatePawn(currentPoint, destination, currentPiece);
+    }
+
+    private void validateMyPiece(Player player, Piece currentPiece) {
+        if (!player.isMyPiece(currentPiece)) {
+            throw new IllegalArgumentException("상대방의 기물을 움직일 수 없습니다.");
+        }
+    }
+
+    private void validateMovablePoint(Point currentPoint, Point destination, Piece currentPiece) {
+        if (!currentPiece.isMovable(currentPoint, destination)) {
+            throw new IllegalArgumentException("해당 기물이 이동할 수 있는 위치가 아닙니다.");
+        }
+    }
+
+    private void validateDestination(Player player, Point destination) {
+        Piece nextPiece = board.get(destination);
+        if (!nextPiece.equals(new Empty(Team.EMPTY)) && player.isMyPiece(nextPiece)) {
+            throw new IllegalArgumentException("이동하려는 위치에 이미 자신의 기물이 있을 수 없습니다.");
+        }
+    }
+
+    private void validateMovableRoute(Point currentPoint, Point destination) {
+        Direction route = currentPoint.findRoute(destination);
+        Point nextPoint = currentPoint.add(route.file(), route.rank());
+        while (!nextPoint.equals(destination)) {
+            if (!board.get(nextPoint).equals(new Empty(Team.EMPTY))) {
```

Comment:
1. new Empty(Team.EMPTY) 는 이 방식으로 밖에 사용 안 될 텐데, 싱글톤으로 만들고 재활용 하는건 어떠세요? Empty.getInstance()
2. NPE를 피하는 습관은 null이 아닌 것이 확실한 걸 앞으로 땡기는 습관입니다. 상수나 null이 아닌 것을 항상 앞에두세요.
```suggestion
            if (!Empty.getInstance().equals(board.get(nextPoint))) {
```

Reply:
1. 리팩토링으로 행과 열, 포인트, 각종 기물들을 캐싱해둘 생각이었어요. 기물들의 경우 하나의 체스판에서는 많이 사용되진않지만 리팩토링 하기 쉽다면 확장성을 고려해도 좋다고 생각이 드네요. 다른 기물들도 잘 적용해보겠습니다!
2. 문자열만 의식해왔었네요. 다른 모든 객체에 대해서 NPE를 피하려고 의식해볼게요.